tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -1029,6 +1029,12 @@ void EnqueueProgramCommand::assemble_device_commands(
                         read_length = max_paged_length_per_sub_cmd;
                         write_length = read_length;
                     }
+                    if (!kernel_bins_dispatch_subcmds.back().empty()) {
+                        auto& back = kernel_bins_dispatch_subcmds.back().back();
+                        if (back.noc_xy_addr != noc_encoding) {
+                            back.flags = CQ_DISPATCH_CMD_PACKED_WRITE_LARGE_FLAG_UNLINK;
+                        }
```

Comment:
To confirm, this is basically unlinking the mcasts for grid 0, when we switch to grid 1. How is this different from the previous case where we unlinked at the bottom of the `kernel_idx` loop?

Reply:
Actually, nvm I read your commit message. This makes sense.